src/context/ShoppingCartContext.tsx

```diff
@@ -0,0 +1,104 @@
+import { createContext, ReactNode, useContext, useState } from "react";
+import { ShoppingCart } from "../components/ShoppingCart";
+import { useLocalStorage } from "../hooks/useLocalStorage";
+import storeItems from "../data/items.json";
+
+type ShoppingCartProviderProps = {
+  children: ReactNode;
+};
+
+type CartItem = {
+  id: number;
+  quantity: number;
+};
+
+type ShoppingCartContext = {
+  openCart: () => void;
+  closeCart: () => void;
+  getItemQuantity: (id: number) => number;
+  increaseCartQuantity: (id: number) => void;
+  decreaseCartQuantity: (id: number) => void;
+  removeFromCart: (id: number) => void;
+  cartQuantity: number;
+  cartItems: CartItem[];
+};
+
+const ShoppingCartContext = createContext({} as ShoppingCartContext);
+
+export function useShoppingCartContext() {
+  const value = useContext(ShoppingCartContext);
+  if (!value)
+    throw new Error(
+      "useShoppingCartContext: undefined value. ShoppingCartProvider must be a parent of this component."
+    );
+  return value;
+}
+
+export function calculateTotal(cartItems: { id: number; quantity: number }[]) {
+  return cartItems.reduce((total, cartItem) => {
+    const item = storeItems.find((i) => i.id === cartItem.id);
+    return total + (item?.price || 0) * cartItem.quantity;
+  }, 0);
+}
+
+export function ShoppingCartProvider({ children }: ShoppingCartProviderProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [cartItems, setCartItems] = useLocalStorage<CartItem[]>(
+    "shopping-cart",
+    []
+  );
+
+  const cartQuantity = cartItems.reduce(
+    (quantity, item) => item.quantity + quantity,
+    0
+  );
+
+  const openCart = () => setIsOpen(true);
+  const closeCart = () => setIsOpen(false);
+
+  function getItemQuantity(id: number): number {
+    return cartItems.find((item) => item.id === id)?.quantity || 0;
+  }
+
+  function increaseCartQuantity(id: number): void {
+    setCartItems((currItems) =>
+      currItems.find((item) => item.id === id)
+        ? currItems.map((item) =>
+            item.id === id ? { ...item, quantity: item.quantity + 1 } : item
+          )
+        : [...currItems, { id, quantity: 1 }]
+    );
```

Comment:
why do you need `find` & `map` both?

Just map should be sufficient - compare the id to update the qty.